### PR TITLE
fix: move update internal deps to worker logic

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes

- Introduce `isInternal` preoperty for type `PackageDependency`
- move update internal deps to worker logic

## Context

- Ref: #15709 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/extract/index.ts [hunk 1]
```diff
@@ -47,6 +47,13 @@ export async function extractAllDependencies(
   const extractResults = await Promise.all(
     extractList.map(async (managerConfig) => {
       const packageFiles = await getManagerPackageFiles(managerConfig);
+      for (const p of packageFiles) {
+        p.deps?.forEach((dep) => {
+          if (!!config.updateInternalDeps && !dep.isInternal) {
```
Comment (rarkins) [inline lib/workers/repository/extract/index.ts:52]:
```suggestion
          if (!config.updateInternalDeps && dep.isInternal) {
```

Shouldn't it be like this? If updateInternalDeps=fasle and it's an internal dep then skip it?

File: lib/workers/repository/extract/index.ts [hunk 2]
```diff
@@ -47,6 +47,13 @@ export async function extractAllDependencies(
   const extractResults = await Promise.all(
     extractList.map(async (managerConfig) => {
       const packageFiles = await getManagerPackageFiles(managerConfig);
+      for (const p of packageFiles) {
+        p.deps?.forEach((dep) => {
```
Comment (rarkins) [inline lib/workers/repository/extract/index.ts:51]: We prefer for.. of instead of forach

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/types.ts [hunk 3]
```diff
@@ -172,6 +172,7 @@ export interface PackageDependency<T = Record<string, any>> extends Package<T> {
   editFile?: string;
   separateMinorPatch?: boolean;
   extractVersion?: string;
+  isInternal?: boolean;
```
Comment (rarkins) [inline lib/modules/manager/types.ts:175]: Also remove `updateInternalDeps` from `ExtractConfig`

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — COMMENTED** [review]
Can you test in a real repo? Not sure if we had any linked from when the capability was added previously

File: renovate.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "updateInternalDeps":"true",
```
Comment (rarkins) [inline renovate.json:3]: Please revert all these package/renovate json changes

**Review by rarkins — CHANGES_REQUESTED** [review]

File: renovate.json (hunk 4 above)
Comment (RahulGautamSingh) [reply renovate.json:3]: > Can you test in a real repo? Not sure if we had any linked from when the capability was added previously

https://github.com/RahulGautamSingh/renovate/pull/2
Is working fine for Renovate.

File: yarn.lock [hunk 5]
```diff
@@ -2043,7 +2043,6 @@
 
 "@renovate/eslint-plugin@https://github.com/renovatebot/eslint-plugin#v0.0.4":
   version "0.0.4"
-  uid "0c444386e79d6145901212507521b8a0a48af000"
```
Comment (rarkins) [inline yarn.lock:2046]: Revert this change

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

File: lib/workers/repository/extract/index.ts [hunk 6]
```diff
@@ -47,6 +47,16 @@ export async function extractAllDependencies(
   const extractResults = await Promise.all(
     extractList.map(async (managerConfig) => {
       const packageFiles = await getManagerPackageFiles(managerConfig);
+      for (const p of packageFiles) {
+        //istanbul ignore if
```
Comment (viceice) [inline lib/workers/repository/extract/index.ts:51]: why we don't have added a test here? 😕

**Review by viceice — COMMENTED** [review]
please add a follow up PR to remove the coverage ignore

File: lib/workers/repository/extract/index.ts (hunk 6 above)
Comment (bobvandevijver) [reply lib/workers/repository/extract/index.ts:51]: I think this actually broke 32.66.6 for us:

```
{
           "name": "renovate",
           "level": 50,
           "logContext": "VeQ76084oIBuT1WliExx-",
           "repository": "intern/symfony-oidc",
           "err": {
             "message": "packageFiles is not iterable",
             "stack": "TypeError: packageFiles is not iterable\n    at /usr/src/app/node_modules/renovate/lib/workers/repository/extract/index.ts:50:23\n    at async Promise.all (index 1)\n    at extractAllDependencies (/usr/src/app/node_modules/renovate/lib/workers/repository/extract/index.ts:47:26)\n    at extract (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:82:20)\n    at extractDependencies (/usr/src/app/node_modules/renovate/lib/workers/repository/process/index.ts:105:26)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:42:52)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:150:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
           },
           "msg": "Repository has unknown error"
         },
```

Comment (viceice) [reply lib/workers/repository/extract/index.ts:51]: `packageFiles` can be null. 

@rarkins please revert this PR, to start again with tests.
